### DISCUSSION TITLE
Mortar method error: Quadrature point out of bounds, truncating

### DISCUSSION THREAD

**Discussion by BoZeng1997** [idaholab/moose#23925] [Q&A Modules: General] [answered]
Hi,

I am running a thermo-mech monolithic problem with mortar method for gap heat transfer. My domain is within 20e-3 around the origin. The deformation is very small. I received an error in the form of
```
Quadrature point: (x,y,z)=(-3.49537e-09,        1,        0) out of bounds, truncating
```
With some slight changes of the input deck (dt, dtmin, total thickness in y direction for the mesh, some material properties), this error always occurs and always with `(x==1||y==1)&z==0` both in parallel and serial computing, in conda environment or with external PETSC. DBG mode did not provide more information.

According to the description of this exception:
https://github.com/idaholab/moose/blob/f92029cda1bc4958facfc5d61d9d5a325423c3f8/framework/src/utils/MortarUtils.C#L258-L261
I output the failed step with `additional_execute_on  = 'FAILED NONLINEAR'` but there was no distorted element.

I need help on how to diagnose and fix this error. Here is the information of my MOOSE version
```
Framework Information:
MOOSE Version:           git commit b35f3b36c9 on 2023-03-23
LibMesh Version:         
PETSc Version:           3.16.6
SLEPc Version:           3.16.2
```

Steps to reproduce this error is in https://github.com/idaholab/moose/issues/23796#issue-1633043012. When the particular issue for that ticket were solved, developers caught the same out of bound error.

**Comment by GiudGiud** [comment]
@lindsayad was involved there

**Comment by lindsayad** [comment]
It looks like @dewenyushu changed this from a warning to an exception in https://github.com/idaholab/moose/pull/19368. @dewenyushu or @recuero have you run across this exception much?

@BoZeng1997 have you considered changing your units?

**Reply by recuero** [reply]
I don't recall running into this. Switching units can potentially resolve these issues.

**Reply by BoZeng1997** [reply]
Do you mean changing the unit so that the values of coordinates are above 1 instead of around 1e-2?

**Reply by lindsayad** [reply, marked as answer]
@BoZeng1997 can you try out https://github.com/idaholab/moose/pull/23954 ?

**Reply by BoZeng1997** [reply]
@lindsayad I tried #23954. The issue never occurred. 
@recuero I also tried changing unit so that the mesh size are of values around 0.1 instead of 1e-4. This also prevents the exception.
I wonder why the unit or the tolerance at finding quadrature point matters in this case (I am assuming that is the idea of #23954). A mesh size of 1e-4 does not seem to be too close to numerical tolerance.
